src/StripeTests/Services/AutoPagingTest.cs

```diff
@@ -287,6 +287,356 @@ public void ListAutoPaging_EndingBefore()
+                .Verify(
+                    "SendAsync",
+                    Times.Once(),
+                    ItExpr.Is<HttpRequestMessage>(m =>
+                        m.Method == HttpMethod.Get &&
+                        m.RequestUri.AbsolutePath == "/v1/pageablemodels" &&
+                        m.RequestUri.Query == "?limit=2&ending_before=pm_127"),
+                    ItExpr.IsAny<CancellationToken>());
+
+            this.MockHttpClientFixture.MockHandler.Protected()
+                .Verify(
+                    "SendAsync",
+                    Times.Once(),
+                    ItExpr.Is<HttpRequestMessage>(m =>
+                        m.Method == HttpMethod.Get &&
+                        m.RequestUri.AbsolutePath == "/v1/pageablemodels" &&
+                        m.RequestUri.Query == "?limit=2&ending_before=pm_125"),
+                    ItExpr.IsAny<CancellationToken>());
+
+            this.MockHttpClientFixture.MockHandler.Protected()
+                .Verify(
+                    "SendAsync",
+                    Times.Once(),
+                    ItExpr.Is<HttpRequestMessage>(m =>
+                        m.Method == HttpMethod.Get &&
+                        m.RequestUri.AbsolutePath == "/v1/pageablemodels" &&
+                        m.RequestUri.Query == "?limit=2&ending_before=pm_123"),
+                    ItExpr.IsAny<CancellationToken>());
+        }
+
+        [Fact]
+        public async Task ListAutoPagingAsync_WithCancellation()
+        {
+            // Set up stubbed requests
+            var response1 = new HttpResponseMessage(HttpStatusCode.OK);
+            response1.Content = new StringContent(GetResourceAsString("pageable_models.0.json"));
+            var response2 = new HttpResponseMessage(HttpStatusCode.OK);
+            response2.Content = new StringContent(GetResourceAsString("pageable_models.1.json"));
+
+            this.MockHttpClientFixture.MockHandler.Protected()
+                .SetupSequence<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.Is<HttpRequestMessage>(m =>
+                        m.Method == HttpMethod.Get &&
+                        m.RequestUri.AbsolutePath == "/v1/pageablemodels"),
+                    ItExpr.IsAny<CancellationToken>())
+                .Returns(Task.FromResult(response1))
+                .Returns(Task.FromResult(response2))
+                .Throws(new StripeTestException("Unexpected invocation!"));
+
+            // Call auto-paging method
+            var service = new PageableService(this.StripeClient);
+            var options = new ListOptions
+            {
+                Limit = 2,
+            };
+            var models = new List<PageableModel>();
+
+            var source = new CancellationTokenSource();
+
+            try
+            {
+                await foreach (var model in service.ListAutoPagingAsync(options).WithCancellation(source.Token))
```

Comment:
This illustrates the new `await foreach` syntax added in C# 8, as well as support for `.WithCancellation()` to pass cancellation tokens.